uaclient/entitlements/base.py

```diff
@@ -1322,5 +1323,12 @@ def _update_sources_list(self):
         if self._is_sources_list_updated:
             return
         event.info(messages.APT_UPDATING_LIST.format(name="standard Ubuntu"))
-        apt.update_sources_list("/etc/apt/sources.list")
+        old_sources_path = "/etc/apt/sources.list"
+        new_sources_path = "/etc/apt/sources.list.d/ubuntu.sources"
+        sources_file = (
+            new_sources_path
+            if os.path.exists(new_sources_path)
+            else old_sources_path
+        )
+        apt.update_sources_list(sources_file)
```

Comment:
Just to avoid code duplication, what about if we make this logic the default `sources_file` we if we don't pass anything to `update_sources_list` ?

Reply:
right
any chance of a situation where we just want to update all sources, pro or not, without specifying a file?
Doesn't exist right now but I wonder if we should account for that

Reply:
Good point, what if we create a shell function that just calls `update_sources_list` with the `/etc/` path ? Then we would still be able to extend the `update_sources_list` function to the use case you mentioned, while also avoiding the code duplication.

What do you think ?

Reply:
I thought that would be overcomplicating it - deduplicated with an apt util function to get the file. Let me know if that suffices

Reply:
That works for me. Thanks @renanrodrigo